packages/eslint-plugin/src/util/misc.ts

```diff
@@ -242,20 +242,19 @@ type NodeWithKey =
 function getStaticMemberAccessValue(
   node: NodeWithKey,
   { sourceCode }: RuleContext<string, unknown[]>,
-): string | undefined {
+): string | null {
   const key =
     node.type === AST_NODE_TYPES.MemberExpression ? node.property : node.key;
   if (!node.computed) {
     return key.type === AST_NODE_TYPES.Literal
       ? `${key.value}`
       : (key as TSESTree.Identifier | TSESTree.PrivateIdentifier).name;
   }
-  const value = getStaticValue(key, sourceCode.getScope(node))?.value as
-    | string
-    | number
-    | null
-    | undefined;
-  return value == null ? undefined : `${value}`;
```

Comment:
seems to be
- a strange inconsistency of JavaScript that `` `${Symbol('a')}` `` throws while `String(Symbol('a'))` and `Symbol('a').toString()` work fine
- a minor type hole in TypeScript that `tsc` _does_ warn for interpolating a `Symbol`, but does _not_ warn for interpolating an `unknown`